main.py

```diff
@@ -106,13 +106,13 @@ async def on_message(message: discord.Message) -> None:
             chosen_team = team_name
 
     if chosen_team:
-        # Add them to the 'verified' role
-        role: discord.Role = discord.utils.get(message.guild.roles, name=VERIFIED_ROLE)
-        await message.author.add_roles(role, reason="A correct password was entered.")
-
         if chosen_team == SPECIAL_TEAM:
             role_name = SPECIAL_ROLE
         else:
+            # Add them to the 'verified' role
+            role: discord.Role = discord.utils.get(message.guild.roles, name=VERIFIED_ROLE)
+            await message.author.add_roles(role, reason="A correct password was entered.")
+            
```

Comment:
Might be worth a comment about why the `SPECIAL_TEAM` users don't need this role.